src/main/java/seedu/hirehub/model/person/ContainsKeywordsPredicate.java

```diff
@@ -56,7 +74,22 @@ public boolean test(Person person) {
             }
             return isPersonTagsContainsTag(person, tags);
         }
-        throw new IllegalStateException("Unexpected value: " + prefix);
+        throw new IllegalStateException("Unexpected prefix: " + prefix);
+    }
+
+    private boolean applicationTest(Application application) {
+        if (prefix.equals(PREFIX_EMAIL)) {
+            Optional<Email> email = (Optional<Email>) keywords;
```

Comment:
Here I am wondering what if the user provides a wrong Prefix but correct input (e.g. e/ John Doe). Isn't the type casting there depending on the fact that the user is always correct with the prefix-input type matching?

Reply:
Correct me if I am wrong plz. Thanks!

Reply:
In SearchApplicationCommandParser, you should see this: new ContainsKeywordsPredicate<>(PREFIX_EMAIL, Optional.of(email)), where the input is parsed into an email (and return an exception if it could not be parsed).

Reply:
Ah okay i see, that pretty much the only edge cases I can think of, which seem to be resolved alr. Thanks!